<?xml version="1.0" encoding="UTF-8"?>

<launch>
	<arg name="drone_name"/>
	<arg name="x" default="0.0"/>
	<arg name="y" default="0.0"/>
	<arg name="z" default="0.0"/> 
	<arg name="slam"/>
	<arg name="height"/>
	
	 <arg name="use_ground_truth_for_tf" default="true"/>
	 <arg name="use_ground_truth_for_control" default="true"/>
	 <arg name="use_pose_estimation" if="$(arg use_ground_truth_for_control)" default="false"/>
	 <arg name="use_pose_estimation" unless="$(arg use_ground_truth_for_control)" default="true"/>

<!-- Spawn a singular hector quadrotor into the environment with SLAM configurations -->
  <group if="$(arg slam)">
  	<!-- Spawn simulated quadrotor uav -->
  	<include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor__slam.launch">
  	    <arg name="name" value="$(arg drone_name)"/>
  	    <arg name="tf_prefix" value="$(arg drone_name)"/>
    	<arg name="x" default="$(arg x)"/>
	    <arg name="y" default="$(arg y)"/>
	    <arg name="use_ground_truth_for_tf" value="$(arg use_ground_truth_for_tf)"/>
	    <arg name="use_ground_truth_for_control" value="$(arg use_ground_truth_for_control)"/>
	    <arg name="use_pose_estimation" value="$(arg use_pose_estimation)"/>
   	    <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_hokuyo_utm30lx.gazebo.xacro"/> 
	    <arg name="world_frame" default="world"/>
	</include>
	
	 <!-- Start rviz visualization with preset config -->
 	 <node pkg="rviz" type="rviz" name="rviz" args="-d $(find swarm)/rviz/$(arg drone_name)_slam.rviz"/>
 	 
	<!-- Launches SLAM --> 
 	 <include file="$(find swarm)/launch/slam/SLAM.launch">
	    <arg name="tf_prefix" value="$(arg drone_name)"/>
	</include>
	
	<!-- Cast broadcaster from world to drone_#/map -->
		<node pkg="tf" type="static_transform_publisher" name="world_to_$(arg drone_name)_tf_broadcaster"  args="0 0 0 0 0 0 /map /$(arg drone_name)/map 100"/>
  </group>

<!-- Spawn a singular hector quadrotor into the environment with Navigation configurations -->
  <group unless="$(arg slam)">
  	<!-- Spawn simulated quadrotor uav -->
  	<include file="$(find hector_quadrotor_gazebo)/launch/spawn_quadrotor_amcl.launch">
  	    <arg name="name" value="$(arg drone_name)"/>
  	    <arg name="tf_prefix" value="$(arg drone_name)"/>
    	<arg name="x" default="$(arg x)"/>
	    <arg name="y" default="$(arg y)"/>
	    <arg name="use_ground_truth_for_tf" value="$(arg use_ground_truth_for_tf)"/>
	    <arg name="use_ground_truth_for_control" value="$(arg use_ground_truth_for_control)"/>
	    <arg name="use_pose_estimation" value="$(arg use_pose_estimation)"/>
   	    <arg name="model" value="$(find hector_quadrotor_description)/urdf/quadrotor_hokuyo_utm30lx.gazebo.xacro"/> 
   		<arg name="world_frame" default="world"/> 
	</include>
   </group>

	<!-- Drone Takeoff Launch -->
  	<include file = "$(find takeoff_land)/launch/takeoff_land.launch">
   		<arg name="tf_prefix" value="$(arg drone_name)"/>
		<arg name="height" value="$(arg height)"/>
   	</include>  
 
</launch>
